wallet/chain/p/builder/builder.go

```diff
@@ -1200,3 +1289,140 @@ func (b *builder) initCtx(tx txs.UnsignedTx) error {
 	tx.InitCtx(ctx)
 	return nil
 }
+
+type spendHelper struct {
+	amountsToBurn  map[ids.ID]uint64
```

Comment:
Are the prefixes necessary? I think the `amounts` (implied by `uint64`) can be removed as it's too generic a word. `s.toBurn` and `s.amountsToBurn` carry equal signal for the reader.

Similarly for use of `amounts` in method variables:

```go
func (s *spendHelper) consumeLockedAsset(assetID ids.ID, amount uint64) uint64 {
	// Stake any value that should be staked
	toStake := min(
		s.toStake[assetID], // Amount we still need to stake
		amount,             // Amount available to stake
	)
	s.toStake[assetID] -= toStake
	return amount - toStake
}
```